### PR TITLE
use warmup steps for lr scheduler, ban steps == -1

### PR DESCRIPTION
**Pull request by wanchaol**
as titled, we don't want to allow steps == -1 case as it would blow up the lr scheduler

### CODE REVIEW

File: train_configs/debug_model.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ lr = 8e-4
 [training]
 batch_size = 8
 seq_len = 2048
-warmup_pct = 0.20  # lr scheduler warm up
+warmup_steps = 5  # lr scheduler warm up
```
Comment (lessw2020) [inline train_configs/debug_model.toml]: nit - don't we want 2 here if the default steps is 10 (i.e. 20% rather than make it 50%).

Comment (tianyu-l) [reply train_configs/debug_model.toml]: Shall we add a comment here suggesting the pct should be around 20%? A user might have no idea how many steps to warm up when changing total number of training steps.

Comment (wanchaol) [reply train_configs/debug_model.toml]: sounds good I'll update. Is 20% the rule of thumb here already? I thought this is more like a tuning thing?

Comment (lessw2020) [reply train_configs/debug_model.toml]: it's definitely a tuning thing - I did 20% to try and be safe bc we are doing such short runs by default (10 steps!) while still showing some reasonable learning gains. 
A more rigourous approach would be 2/(1−β2) training iterations for the warmup with the current linear scheduling, which is 2000 iterations assuming default AdamW.  (see https://arxiv.org/abs/1910.04209). 

The issue here is we want to show a reasonable curve in the short term and doing that in 10 steps or even 100 is very different than the more rigorous algo above (which can be simplified to 2k warmup iterations) or even plugged in to the lr if we wanted to get fancy and be more exact.

**Pull request merged**